src/plugins/intel_cpu/src/cpu_tensor.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <map>
+
+#include "cpu_tensor.h"
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+//For the beginning probably one contructor should be enough
+Tensor::Tensor(MemoryPtr memptr) : m_memptr{memptr}, m_strides{}, m_strides_once{} {
+    OPENVINO_ASSERT(m_memptr != nullptr);
+
+    // only support plain data format ncsp.
+    auto blocked_desc = m_memptr->getDescWithType<BlockedMemoryDesc>();
+    OPENVINO_ASSERT(blocked_desc, "intel_cpu::Tensor only supports blocked memory descriptor.");
+    OPENVINO_ASSERT(blocked_desc->hasLayoutType(LayoutType::ncsp), "intel_cpu::Tensor only supports memory with ncsp layout.");
+
+    m_element_type = InferenceEngine::details::convertPrecision(blocked_desc->getPrecision());
+    m_shape = blocked_desc->getShape().toPartialShape().to_shape();
```

Comment:
What if the underlying memory object descriptor is changed after the `Tesnor` object has been instanced? Then the shape change is not reflected by the `Tensor` object. Suggest moving this lines to the corresponding methods. This wrapped doesn't solely own the memory object, and it may be modified somewhere outside the object as well. The same is true for the strides.

Reply:
done

Reply:
The problem is still here. What if the underlying memory object memory descriptor was changed in a separate code frame? How the shape and the strides are updated. The `Tensor` object stores only a reference to the memory, thus the memory itself is not protected from modifications outside the `Tensor` object. Also this indicates  a lack of tests. Simple test could be as follows:

```cpp
    Memory mem(eng, desc);
    Tensor tensor(mem);
    ASSERT_EQ(mem.getDesc().getStrides() * data_size, tensor.get_strides()); // here we assume desc.getStrides() * data_size is a scaling of the whole vector
   auto desc2 = desc.cloneWithNewDims(newShape);
   mem.redefineDesc(desc2);
   ASSERT_EQ(mem.getDesc().getStrides() * data_size, tensor.get_strides()); // this check fails since the current implementation doesn't reflect the change of the underlining memory object state
```

Reply:
fixed and updated with more unit tests.